src/dbusmenuexporter.cpp

```diff
@@ -55,11 +55,6 @@ int DBusMenuExporterPrivate::idForAction(QAction *action) const
 
 void DBusMenuExporterPrivate::addMenu(QMenu *menu, int parentId)
 {
-    if (menu->findChild<DBusMenu *>()) {
-        // This can happen if a menu is removed from its parent and added back
-        // See KDE bug 254066
-        return;
-    }
```

Comment:
I know nothing about KDE bug 254066, but this check seems reasonable to me. Shouldn't it be replaced by another check that's compatible with the current change?

Reply:
I believe no. The above and the fact, that all DBus exporter objects were tied to (made children of) the exported `QMenu` objects was wrong, IMO.
-  One `QMenu` can be exported by as many DBus exporters as the user decides to do and the user is responsible for releasing the DBus exporter, when it's not needed any more. Not to rely just on the lifespan of exported `QMenu` (the `destroyed()` signal of `QMenu` is of course watched and acted accordingly). The exported `QMenu` can be e.g. some kind of _main menu_ exported just for short periods of time, when needed (why should the DBus exporter force to build a dedicated menu?).
- The DBus exporter objects (`DBusMenu`, ..) should not be left alive after the main `DBusMenuExporter` object as they are of no use afterwards.

Reply:
Thanks; it makes sense.

This and other related changes could be tested/discussed/merged after LXQt 2.2.0.